src/SonarScanner.MSBuild.PreProcessor/WebClientDownloaderBuilder.cs

```diff
@@ -75,13 +77,45 @@ public WebClientDownloaderBuilder AddCertificate(string clientCertPath, string c
             return this;
         }
 
-        handler = new HttpClientHandler();
+        handler ??= new HttpClientHandler();
         handler.ClientCertificateOptions = ClientCertificateOption.Manual;
         handler.ClientCertificates.Add(new X509Certificate2(clientCertPath, clientCertPassword));
 
         return this;
     }
 
+    public WebClientDownloaderBuilder AddServerCertificate(string serverCertPath, string serverCertPassword)
+    {
+        if (serverCertPath is null || serverCertPassword is null)
+        {
+            return this;
+        }
+
+        handler ??= new();
+        trustStore ??= new();
+        trustStore.Import(serverCertPath, serverCertPassword, X509KeyStorageFlags.DefaultKeySet);
+        handler.ServerCertificateCustomValidationCallback = (message, certificate, chain, errors) =>
+            ServerCertificateCustomValidationCallback(this.trustStore, message, certificate, chain, errors);
+
+        return this;
+    }
+
+    private static bool ServerCertificateCustomValidationCallback(X509Certificate2Collection trustStore, HttpRequestMessage message, X509Certificate2 certificate, X509Chain chain, SslPolicyErrors errors)
+    {
+        if (errors == SslPolicyErrors.None)
+        {
+            return true;
+        }
+        if (errors == SslPolicyErrors.RemoteCertificateChainErrors)
+        {
+            if (trustStore.Find(X509FindType.FindBySerialNumber, certificate.SerialNumber, validOnly: false) is { Count: > 0 } certificatesInTrustStore)
```

Comment:
A first implementation of the validation. The remaining parts will be implemented test driven along the list in https://sonarsource.atlassian.net/browse/SCAN4NET-209